fedimint-api-client/src/api/iroh.rs

```diff
@@ -238,6 +200,131 @@ impl IrohConnector {
         self
     }
 
+    pub fn node_id_from_url(url: &SafeUrl) -> anyhow::Result<NodeId> {
+        if url.scheme() != "iroh" {
+            bail!(
+                "Unsupported scheme: {}, passed to iroh endpoint handler",
+                url.scheme()
+            );
+        }
+        let host = url.host_str().context("Missing host string in Iroh URL")?;
+
+        let node_id = PublicKey::from_str(host).context("Failed to parse node id")?;
+
+        Ok(node_id)
+    }
+}
+
+#[async_trait::async_trait]
+impl crate::api::Endpoint for IrohEndpoint {
+    async fn connect(
+        &self,
+        url: &SafeUrl,
+        api_secret: Option<&str>,
+    ) -> PeerResult<DynClientConnection> {
+        if api_secret.is_some() {
+            // There seem to be no way to pass secret over current Iroh calling
+            // convention
+            PeerError::Connection(anyhow::format_err!(
+                "Iroh api secrets currently not supported"
+            ));
+        }
+        let node_id = Self::node_id_from_url(url).map_err(PeerError::InvalidEndpoint)?;
+        let mut futures = FuturesUnordered::<
+            Pin<Box<dyn Future<Output = (PeerResult<DynClientConnection>, &'static str)> + Send>>,
+        >::new();
+        let connection_override = self.connection_overrides.get(&node_id).cloned();
+
+        // Use connection pool for stable endpoint
+        let self_clone = self.clone();
+        futures.push(Box::pin({
+            let connection_override = connection_override.clone();
+            async move {
+                (
+                    self_clone
+                        .get_or_create_connection_stable(node_id, connection_override)
```

Comment:
If we are re-implementing connection reuse everywhere, why not lift it to a higher layer as originally envisioned? See previous comment about WS connections. 

There is no use case for connection reuse between clients, the thing we want to reuse is the connector/some low-level connections like client-relay connections, but not high-level client-server ones.

Maybe this is just an intermediate step?